sdk/vaa/structs.go

```diff
@@ -438,6 +438,33 @@ func (v *VAA) VerifySignatures(addresses []common.Address) bool {
 
 	return true
 }
+func (v *VAA) Verify(addresses []common.Address) bool {
+	if addresses == nil {
+		return false
+	}
+
+	// Check if VAA doesn't have any signatures
+	if len(v.Signatures) == 0 {
+		return false
+	}
+
+	// Verify VAA has enough signatures for quorum
+	quorum := CalculateQuorum(len(addresses))
+	if len(v.Signatures) < quorum {
+		return false
+	}
+
+	// Verify VAA signatures to prevent a DoS attack on our local store.
+	if !v.VerifySignatures(addresses) {
+		return false
+	}
+
+	return true
+}
+
+func CalculateQuorum(numGuardians int) int {
```

Comment:
Are you going to make the processor code use these routines from the sdk? Otherwise we are going to have the quorum calculation in multiple places, which I don't think is good.

Reply:
Fixed.